backend/lib/endpoints/organizations.js

```diff
@@ -0,0 +1,95 @@
+const httpErrors = require("http-errors");
+
+const {
+  createOrganizationSchema,
+  deleteOrganizationSchema,
+  getOrganizationSchema,
+  getOrganizationsSchema,
+  updateOrganizationSchema,
+} = require("./schema/organizations");
+
+/*
+ * /api/organizations
+ */
+async function routes(app) {
+  const Organization = app.mongo.model("Organization");
+
+  app.delete(
+    "/:organizationId",
+    {
+      preValidation: [app.authenticate],
+      schema: deleteOrganizationSchema,
+    },
+    async (req) => {
+      // TODO: make sure user can only delete organizations they own
+      return Organization.findByIdAndRemove(req.params.organizationId);
+    },
+  );
+
+  app.get(
+    "/",
+    {
+      preValidation: [app.authenticate],
+      schema: getOrganizationsSchema,
+    },
+    async (req, reply) => {
+      const { ownerId } = req.params;
+      const filter = ownerId ? { ownerId } : {};
+      const sortedOrganizations = await Organization.find(filter).sort({
+        name: 1,
+      });
+      return reply.send(sortedOrganizations);
+    },
+  );
+
+  app.get(
+    "/:organizationId",
+    {
+      preValidation: [app.authenticate],
+      schema: getOrganizationSchema,
+    },
+    async (req) => {
+      const result = await Organization.findById(req.params.organizationId);
+      if (result === null) {
+        return new httpErrors.NotFound();
+      }
+      return result;
+    },
+  );
+
+  app.patch(
+    "/:organizationId",
+    {
+      preValidation: [app.authenticate],
+      schema: updateOrganizationSchema,
+    },
+    async (req, reply) => {
+      // TODO: make sure user can only update organizations they own
+      const organization = await Organization.findById(
+        req.params.organizationId,
+      );
+      if (organization === null) {
+        return reply.send(new httpErrors.NotFound());
+      }
+      Organization.schema.eachPath((key) => {
```

Comment:
@robinv85 something I noticed when I had the logic similar to what was in the the `posts#patch` endpoint...any new keys that are set in the patch operation were skipped over and not saved to the document. So I had to alter the logic here to iterate over each attribute in the `Organization` model, and check that the attribute key exists in the request body. Just wanted to give you a heads up, so that we should fix it in the `posts` endpoint as well.

Since this pattern will be used by all patch operations, perhaps we could add a reusable function in `utils.js`?

Reply:
Actually, I just realized that the issue was that the code in `posts#patch` is just checking that the key exists in the BSON document. Removing that check should be sufficient. See my latest [commit](https://github.com/manualzuru/FightPandemics.com/pull/105/commits/3d8766a1486d9daaedf44450b1d14e0f57c0a4e5).

Reply:
yeah commented on that link